src/T8code.jl

```diff
@@ -157,6 +159,84 @@ not a system-provided `t8code` installation. In this case, T8code.jl is not usab
 preferences_set_correctly() = !(_PREFERENCE_LIBT8 == "t8code_jll" &&
                                 MPIPreferences.binary == "system")
 
+# Minimal reference tracker holding active t8code related objects
+# created throughout the life time of a Julia session. t8code objects
+# should remove themselves from the tracker when they get finalized.
+if !@isdefined(T8CODE_OBJECT_TRACKER)
+    T8CODE_OBJECT_TRACKER = Dict()
+end
+
+"""
+    T8codeForestWrapper
+
+Lightweight `t8_forest_t` pointer wrapper which helps to free
+resources allocated by t8code in an orderly fashion.
+
+When initialized with a t8code forest pointer the wrapper
+registers itself with a t8code object tracker called `__T8CODE_OBJECT_TRACKER`.
+
+In serial mode the wrapper and in consequence the t8code forest
+can be finalized immediately whenever Julia's garbage collector sees fit.
+
+In (MPI) parallel mode the wrapper (and the t8code forest) is kept till the end
+of the session or when finalized explicitly. At the end of the session (resp.
+when the program shuts down) the object tracker finalizes all registered t8code
+objects in sync with all MPI ranks. This is necessary since t8code internally
+allocates MPI shared arrays. See `src/auxiliary/t8code.jl` for the finalization
+code when Trixi is shutting down.
+"""
+mutable struct ForestWrapper
+    pointer::Ptr{t8_forest} # cpointer to t8code forest
+    unique_id::UInt
```

Comment:
```suggestion
    unique_id::UInt64
```
We have `UInt === Uint64`, but I think `UInt64` is more explicit.

Reply:
Agreed.